TD-2051 New Prop and component conversion to typescript

m4manjesh opened the pull request:
<!-- Insert YouTrack link if relevant -->

Contributes to [TD-2051](https://sce.myjetbrains.com/youtrack/issue/TD-2051/Improve-the-file-validation-for-images-in-MODEL-and-SCENE)

As a part of the bug fix, we have decided to improve the display of error messages for uploaders. In a few cases, this was handled inside apps, but the library we used to build the component already has an option to enable alerts. Here we are just enabling the `showAlerts` by passing them as a prop. By default, this will be enabled and all the error messages showing on a snack bar will be handled by the component itself. These changes are applied for both `ImageUplaoder` and `FileUploader`.
As a part of this, i have converted these components to typescript as well.

## Changes

- New `showErrorAlert` props added
- Components converted to typescript.

## UI/UX

ImageUploader
![Screenshot (259)](https://github.com/IPG-Automotive-UK/react-ui/assets/56511816/66e2fdaf-593c-48a2-a24f-81bff902177b)

FileUploader
![Screenshot (258)](https://github.com/IPG-Automotive-UK/react-ui/assets/56511816/60311322-48f6-4bf1-8c2a-da54f62a96fb)

## Testing notes

- With the enabled `showErrorAlert` prop if you pass wrong or multiple files should show an error message on a snackbar.

## Author checklist before assigning a reviewer

- [x] Reviewed my own code-diff.
- [ ] Branch has been run in docker.
- [x] PR assigned to me or an appropriate delegate.
- [x] Relevant labels added to the PR.
- [ ] Appropriate tests have been added.
- [x] Lint and test workflows pass.

mattcorner reviewed (approved):
I fixed up some of the typescript & styling issues as it was quicker to do so after I'd investigated what was going on. I also took @Sowbhagya-ipg through the UI behaviour as you hadn't. 

Other than that, LGTM.

m4manjesh commented:
> I fixed up some of the typescript & styling issues as it was quicker to do so after I'd investigated what was going on. I also took @Sowbhagya-ipg through the UI behaviour as you hadn't.
> 
> Other than that, LGTM.



> I fixed up some of the typescript & styling issues as it was quicker to do so after I'd investigated what was going on. I also took @Sowbhagya-ipg through the UI behaviour as you hadn't.
> 
> Other than that, LGTM.

Thanks!
